src/handlers/thirdparty.ts

```diff
@@ -34,4 +37,61 @@ export function handleThirdPartyAdded(event: ThirdPartyAdded): void {
 
   let metric = buildCountFromThirdParty()
   metric.save()
-}
\ No newline at end of file
+}
+
+export function handleItemUpdated(event: ItemUpdated): void {
+  let itemEntityId = buildItemEntityId(
+    event.params._thirdPartyId,
+    event.params._itemId
+  )
+  let item = Item.load(itemEntityId)
+  if (item == null) {
+    log.error('An item with a non existent id "{}" tried to be updated', [
+      itemEntityId
+    ])
+    return
+  }
+
+  item.rawMetadata = event.params._metadata
+  let metadata = buildMetadata(item.id, item.rawMetadata)
+  item.metadata = metadata.id
+
+  item.save()
+}
+
+export function handleItemAdded(event: ItemAdded): void {
+  let collectionId = getCollectionId(event.params._itemId)
+  let itemId = getItemId(event.params._itemId)
+  if (collectionId == null || itemId == null) {
+    log.error(
+      'An item was added in the TPR "{}" with an incorrect id - collectionId: "{}", itemId: "{}"',
+      [
+        event.params._thirdPartyId,
+        collectionId ? collectionId : 'null',
+        itemId ? itemId : 'null'
+      ]
+    )
+    return
+  }
+
+  let item = Item.load(
+    buildItemEntityId(event.params._thirdPartyId, event.params._itemId)
+  )
+  if (item === null) {
+    item = new Item(event.params._itemId)
```

Comment:
I guess that it would be a great idea to use the result of `buildItemEntityId` here too